packages/events/src/service/events/events.ts

```diff
@@ -211,6 +233,12 @@ export async function addAction(
     if (!(await fileExists(fileValue.data.filename, token))) {
       throw new Error(`File not found: ${fileValue.data.filename}`)
     }
+    await deleteSpecificFieldAttachments(
+      token,
+      event,
+      key,
+      value as FileFieldValue
+    )
```

Comment:
Sorry, can you explain what attachments this code deletes exactly? I'm expecting to see checks to ensure we only delete files that were previously part of drafts of the action but are not referred to anymore now when the action will be committed.

As tempting it is to do this as part of the logic that checks if all action files exists, let's still separate this procedure completely from the file checking logic to a clearly named function like 

```ts
function cleanUnreferencedAttachmentsFromPreviousDrafts(event, token)
```